drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -641,7 +699,7 @@ void parseSDFLink(RigidBodySystem& sys, string model_name, XMLElement* node,
                            transform_link_to_model);
     }
 
-    if (type.compare("ray") == 0) {
+    if (type == "ray") {
```

Comment:
Do you want to add an `else` clause that outputs some warning or error about not handling sensors than than 'ray'?